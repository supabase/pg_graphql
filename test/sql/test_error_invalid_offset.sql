begin;

    create table blog(
        id serial primary key,
        owner_id integer not null,
        name varchar(255) not null,
        description text
    );

    insert into blog(owner_id, name, description)
    values
        (1, 'A: Blog 1', 'first'),
        (2, 'A: Blog 2', 'second');


    select graphql.resolve($$
        {
          blogCollection(first: -1) {
            edges {
              cursor
              node {
                ownerId
              }
            }
          }
        }

    $$);

    select graphql.resolve($$
        {
          blogCollection(last: -1) {
            edges {
              cursor
              node {
                ownerId
              }
            }
          }
        }

    $$);

    comment on schema public is E'@graphql({"max_rows": -1, "inflect_names": true})';

    select graphql.resolve($$
        {
          blogCollection {
            edges {
              cursor
              node {
                ownerId
              }
            }
          }
        }

    $$);


rollback;
