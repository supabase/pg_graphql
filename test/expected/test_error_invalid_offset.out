begin;
    create table blog(
        id serial primary key,
        owner_id integer not null,
        name varchar(255) not null,
        description text
    );
    insert into blog(owner_id, name, description)
    values
        (1, 'A: Blog 1', 'first'),
        (2, 'A: Blog 2', 'second');
    select graphql.resolve($$
        {
          blogCollection(first: -1) {
            edges {
              cursor
              node {
                ownerId
              }
            }
          }
        }

    $$);
                                    resolve                                     
--------------------------------------------------------------------------------
 {"data": null, "errors": [{"message": "`first` must be an unsigned integer"}]}
(1 row)

    select graphql.resolve($$
        {
          blogCollection(last: -1) {
            edges {
              cursor
              node {
                ownerId
              }
            }
          }
        }

    $$);
                                    resolve                                    
-------------------------------------------------------------------------------
 {"data": null, "errors": [{"message": "`last` must be an unsigned integer"}]}
(1 row)

    comment on schema public is E'@graphql({"max_rows": -1, "inflect_names": true})';
    select graphql.resolve($$
        {
          blogCollection {
            edges {
              cursor
              node {
                ownerId
              }
            }
          }
        }

    $$);
                                                           resolve                                                            
------------------------------------------------------------------------------------------------------------------------------
 {"errors": [{"message": "Error while loading schema, check comment directives. invalid value: integer `-1`, expected u64"}]}
(1 row)

rollback;
